### PR TITLE
Fix `unit.returners.test_local_cache` for Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fixes failing unit tests:
```
unit.returners.test_local_cache.LocalCacheCleanOldJobsTestCase.test_clean_old_jobs_empty_jid_dir_remains
unit.returners.test_local_cache.LocalCacheCleanOldJobsTestCase.test_clean_old_jobs_empty_jid_dir_removed
```

Fixes Windows paths

Gets the current time after it gets the creation date/time. Getting the current time at the top of the function was returning a current time that was earlier than the creation time of the directory which caused the `hours_difference` to be a negative number.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt-jenkins/issues/439

### CODE REVIEW

File: tests/unit/returners/test_local_cache.py [hunk 1]
```diff
@@ -72,6 +72,12 @@ def test_clean_old_jobs_empty_jid_dir_removed(self):
         # Make sure there are no files in the directory before continuing
         self.assertEqual(jid_file, None)
 
+        # Windows needs some time to release the file handles to the new temp
+        # dir before trying to clean the job cache to avoid a race condition
```
Comment (cachedout) [inline tests/unit/returners/test_local_cache.py]: If any call in that function isn't properly cleaning up a filehandle, that's a serious bug. We need to investigate why and solve that issue.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by twangboy** [comment]
This PR can't be merged until we can show that `local_cache.clean_old_jobs` doesn't support clean times < 1s accuracy on certain platforms (Windows) and is documented.

**Comment by twangboy** [comment]
@cachedout What do you think of this fix?

**Comment by cachedout** [comment]
I think this makes sense. We can probably just move the `time.time()` call into the line where the subtraction is actually done though. No need to make an assignment on a line before, that I can see.

**Comment by twangboy** [comment]
@cachedout Done

**Review by rallytime — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Pull request merged**